Makefile

```diff
@@ -161,8 +161,31 @@ sub-image-%:
 	@$(MAKE) ARCH=$* image
 
 image: ## make the image for a single ARCH
-	docker buildx build --load -t $(BUILD_IMAGE):latest-$(ARCH) -f Dockerfile --platform $(OS)/$(ARCH) .
-	echo "Done. image is at $(BUILD_IMAGE):latest-$(ARCH)"
+	docker buildx build --load -t $(BUILD_IMAGE):$(BUILD_TAG)-$(ARCH) -f Dockerfile --platform $(OS)/$(ARCH) .
+	echo "Done. image is at $(BUILD_IMAGE):$(BUILD_TAG)-$(ARCH)"
+
+image-all-push: $(addprefix sub-image-push-, $(ARCHES)) ## Push all built images.
+sub-image-push-%:
+	@$(MAKE) ARCH=$* image-push
+
+image-push: ## Push image to registry for a single ARCH.
+	docker push $(BUILD_IMAGE):$(BUILD_TAG)-$(ARCH)
+
+image-manifest-all: image-manifest-create image-manifest-annotate-all ## Make single image manifest for all supported ARCH.
```

Comment:
Why is there `manifest-all` and `manifest-create`? Since `-all` is for "all arches", isn't `manifest` by definition for "all"?

Reply:
I think you can simplify the multiple `manifest` targets into fewer. Just have a single one that creates a manifest for all know arches.

Reply:
>Why is there manifest-all and manifest-create? Since -all is for "all arches", isn't manifest by definition for "all"?

This is because order of operation is important here. It must be:
1. Ensure all images are pushed.
2. Create the manifest.
3. Annotate it for all architectures.

I tried to simplify it, please have a look :)

Reply:
I actually never have used `docker manifest create` and `docker manifest annotate`; I have been using manifest-tool since day 1: just push a bunch of images, and then create an index and push it in a single command.

Unfortunately, the documentation for `docker manifest create` and `docker manifest annotate` are sparse (to put it mildly). 

I would have thought `manifest create` makes it locally, and `manifest annotate adds data to the specific entry in the manifests list in the index, and `manifest push` pushes it. But if I try to do `manifest create`, I get registry related errors?

Either way, is there any reason for all of the targets? Why not just `make manifest` and have it create, annotate and push? If someone wants to do something manually other than that, they can.

Reply:
>I would have thought manifest create makes it locally, and manifest annotate adds data to the specific entry in the manifests list in the index, and manifest pushpushes it. But if I try to domanifest create`, I get registry related errors?

Yeah, this is how it works. However, when creating the manifest, I think it takes the image information from the registry, as manifest only contains a references to other images, which must be already pushed.

>Either way, is there any reason for all of the targets? Why not just make manifest and have it create, annotate and push? If someone wants to do something manually other than that, they can.

So `annotate` is a separate target so we can use `addprefix` expansion pattern which is used for building the image. But creating the manifest must create before, hence separate target to run it.

And I think `push` target can be merged with `all`, as running it will push individual images anyway.

If you could help me with simplifying it, that would be great, as I'm not sure how passing ARCH works exactly etc.

Reply:
Eh, I just tried and could not come up with a cleaner way. Minutely, but not seriously. I dislike having it in the Makefile _and_ in the GitHub Actions. But so be it.

Let's remove the `image-*` prefixes to make it easier to understand and take it as is.

Reply:
>Eh, I just tried and could not come up with a cleaner way. Minutely, but not seriously. I dislike having it in the Makefile and in the GitHub Actions. But so be it.

This is why I dislike GitHub Actions a bit. IMO CI should only automate tasks which can be executed locally. So if your CI breaks, you can still perform all maintenance tasks.

I removed the `image-*` prefixes and merged commits. I'm testing now and if everything works fine, will push and this will be ready to merge.

Thanks again for looking into that!

Reply:
@invidian Have you tried [`act`](https://github.com/nektos/act)?

I run GH Actions locally with: `act -P ubuntu-latest=nektos/act-environments-ubuntu:18.04 -b` 

I've run into some Actions that don't work as expected, most recently with local mount points from the Mac needing to get passed into actions that invoke `docker`.